### DISCUSSION TITLE
New branch default name hook?

### DISCUSSION THREAD

**Discussion by xendk** [magit/magit#4663] [Support] [answered]
Is there a way to pre-populate the new branch name prompt (`b c`)? My personal use case is to pre-populate it with issue ID prefix if Emacs can sniff it from my running Harvest timer. Basically the same behaviour as when `b c`ing `origin/something` and magit suggesting `something` as branch name.

I guess creating a new command for it is an alternative, but I'd rather keep my create-branch-muscle-memory.

**Comment by kyleam** [comment, marked as answer]
> Is there a way to pre-populate the new branch name prompt (`b c`)?

No, there's no customization to hook into.  Here's the code responsible for that:

https://github.com/magit/magit/blob/3cfc8458e14c705afdfbeb7dcd3d3f43d7479344/lisp/magit-branch.el#L395-L415

One option would be to advise `magit-branch-read-args`, something like:

```elisp
(defun my/issue-number ()
  (let ((num (random 999)))
    (and (/= (% num 3) 0)
         (format "issue-%03d" num))))

(defun my/magit-read-branch-args-wrapper (fn prompt &optional default-start)
  (if-let (issue (my/issue-number))
      (let ((branch (magit-read-string-ns (concat prompt " named") issue)))
        (list branch (magit-read-starting-point prompt branch default-start)))
    (funcall fn prompt default-start)))

(advice-add 'magit-branch-read-args :around
            #'my/magit-read-branch-args-wrapper
            '((name . "magit-read-branch-args/issue-number-default")))
```

That's a bit invasive and may require you make tweaks as things change on Magit's side, though.

**Reply by xendk** [reply]
A tad involved, and I guess it only gets worse if one would like it to only provide a default if magit has no other suggestions.

Smells like a feature request, but there's probably a lot of details I haven't thought of.
